--- check_disk_btrfs	2018-07-30 16:17:44.400000000 -0500
+++ check_disk_btrfs.new	2018-07-30 16:18:54.140000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # COPYRIGHT:
 #
 # This software is Copyright (c) 2015 NETWAYS GmbH, Michael Friedrich
@@ -54,7 +54,7 @@
 
 VERSION = '2.1.1'
 sudo_path = "/usr/bin/sudo"
-btrfs_path = "/usr/sbin/btrfs"
+btrfs_path = "/sbin/btrfs"
 verbose = True
 
 class BtrfsCommandError(Exception):
