--- pyunifi-2.14/pyunifi/controller.py.old	2023-01-03 06:17:00.809961290 -0600
+++ pyunifi-2.14/pyunifi/controller.py	2023-01-03 06:17:20.673294041 -0600
@@ -79,7 +79,7 @@
         self.ssl_verify = ssl_verify
 
         if ssl_verify is False:
-            warnings.simplefilter("default", category=requests.packages.
+            warnings.simplefilter("ignore", category=requests.packages.
                                   urllib3.exceptions.InsecureRequestWarning)
 
         self.session = requests.Session()
